tests/security/tpm2/tpm2_env_setup.pm

```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2020 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: TPM2 test environment prepare
+#          Install required packages and create user, start the abrmd service
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#64899, tc#1742297, tc#1742298
+
+use strict;
+use warnings;
+use base 'opensusebasetest';
+use testapi;
+use utils 'zypper_call';
+use power_action_utils "power_action";
+use version_utils qw(is_opensuse is_sle is_leap is_tumbleweed is_storage_ng);
+
+sub run {
+    my ($self) = @_;
+    select_console "root-console";
+
+    # Add user tss, tss is the default user to start tpm2.0 service
+    assert_script_run "useradd -d /home/tss -m tss";
+    assert_script_run "echo tss:susetesting | chpasswd";
+
+    # Install the tpm2.0 related packages
+    # and then start the TPM2 Access Broker & Resource Manager
+    if (is_sle('>=15-SP2')) {
+        zypper_call "in ibmswtpm2 tpm2.0-abrmd tpm2.0-abrmd-devel openssl tpm2-0-tss tpm2-tss-engine tpm2.0-tools";
+    }
+    else {
+        die "Unsupported system version, SLES15SP2 and above are supported\n";
```

Comment:
if not support this logic should put into main_common.pl, only load this test module if is_sle('>=15-SP2')

Reply:
Thanks Wei,
I have modified my code and add the version check to main_common